wit/types.wit

```diff
@@ -0,0 +1,128 @@
+// The `wasi:http/types` interface is meant to be imported by components to
+// define the HTTP resource types and operations used by the component's
+// imported and exported interfaces.
+default interface types {
+  
+  // This type corresponds to HTTP standard Methods.
+  variant method {
```

Comment:
Should an additional value taking a `string` parameter, like the `other(string)` value in `scheme`, be added in order to support [extension methods](https://datatracker.ietf.org/doc/html/rfc2616#section-5.1.1) used for example in [WebDAV](http://www.webdav.org/specs/rfc4918.html#http.methods.for.distributed.authoring)?

Reply:
Ah, good point!  [Added](https://github.com/WebAssembly/wasi-http/pull/3/commits/67f6b2f016524a407026a68ddfd218d879aefa0f)